runtime/sema/errors.go

```diff
@@ -2929,6 +2929,46 @@ func (e *InvalidAssignmentAccessError) SecondaryError() string {
 	)
 }
 
+// UnauthorizedReferenceAssignmentError
+
+type UnauthorizedReferenceAssignmentError struct {
+	RequiredAccess [2]Access
```

Comment:
Why are exactly two accesses needed here?

Reply:
Currently, to assign to an array/dictionary element, one would need to have either an `auth(Mutate)` reference or an `auth(Insert, Remove)` reference. So the required two accesses are that.